lib/rspec-puppet/cache.rb

```diff
@@ -11,10 +11,10 @@ def initialize(&default_proc)
     end
 
     def get(*args, &blk)
+      expire!
       if !@cache.has_key? args
         @cache[args] = (blk || @default_proc).call(*args)
         @lra << args
-        expire!
```

Comment:
That is some serious caching stuff. To deep for me to follow on Friday evening 😉